common/src/Common/GistStore.hs

```diff
@@ -0,0 +1,22 @@
+{-# LANGUAGE DeriveGeneric #-}
+module Common.GistStore where
+
+import Generics.Deriving.Monoid    (mappenddefault)
+import GHC.Generics (Generic)
+import Data.Aeson (ToJSON, FromJSON)
+import Data.Text (Text)
+
+-- | Meta data about a gist. (Like name and description)
+data GistMeta = GistMeta
+  { _gistMeta_fileName    :: Text
+    -- ^ What filename to use in the gist.
+  , _gistMeta_description :: Text
+    -- ^ The description of the gist to create.
+  }
+  deriving (Show, Generic)
+
+-- TODO Remove these derived instances
```

Comment:
Why? When?

Reply:
Because the derived instances are fragile to making incompatible changes to the json structures accidentally. We have a policy now to not derive json instances.

It probably should have already been done, but gists don't work on desktop anyway.

Reply:
Ah nevermind, I thought the comment also applied to `Semigroup`.